tests/pipeline/test_events.py

```diff
@@ -1,299 +1,425 @@
+            end_date=self.trading_days[-1],
+        )
 
-class BlazeEventDataSetLoader(BlazeEventsLoader):
-    concrete_loader = EventDataSetLoader
-    _expected_fields = frozenset({ANNOUNCEMENT_FIELD_NAME,
-                                  TS_FIELD_NAME,
-                                  SID_FIELD_NAME})
-
-    def __init__(self,
-                 expr,
-                 dataset=EventDataSet,
-                 **kwargs):
-        super(
-            BlazeEventDataSetLoader, self
-        ).__init__(expr,
-                   dataset=dataset,
-                   **kwargs)
+        for c in EventDataSet.columns:
+            if c in self.next_value_columns:
+                self.check_next_value_results(c, results[c.name].unstack())
+            elif c in self.previous_value_columns:
+                self.check_previous_value_results(c, results[c.name].unstack())
+            else:
+                raise AssertionError("Unexpected column %s." % c)
+
+    def check_previous_value_results(self, column, results):
+        """
+        Check previous value results for a single column.
+        """
+        events = self.raw_events
+        # Remove timezone info from trading days, since the outputs
+        # from pandas won't be tz_localized.
+        dates = self.trading_days.tz_localize(None)
+        for asset, asset_result in results.iterkv():
+            relevant_events = events[events.sid == asset.sid]
+            self.assertEqual(len(relevant_events), 2)
+
+            v1, v2 = relevant_events[self.previous_value_columns[column]]
+            event1_first_eligible = max(
+                # .ix doesn't work here because the frame index contains
+                # integers, so 0 is still interpreted as a key.
+                relevant_events.iloc[0].loc[['event_date', 'timestamp']],
+            )
+            event2_first_eligible = max(
+                relevant_events.iloc[1].loc[['event_date', 'timestamp']]
+            )
+
+            for date, computed_value in zip(dates, asset_result):
+                if date >= event2_first_eligible:
+                    # If we've seen event 2, it should win even if we've seen
+                    # event 1, because events are sorted by event_date.
+                    self.assertEqual(computed_value, v2)
+                elif date >= event1_first_eligible:
+                    # If we've seen event 1 but not event 2, event 1 should
+                    # win.
+                    self.assertEqual(computed_value, v1)
+                else:
+                    # If we haven't seen either event, then we should have
+                    # column.missing_value.
+                    assert_equal(
+                        computed_value,
+                        column.missing_value,
+                        # Coerce from Timestamp to datetime64.
+                        allow_datetime_coercions=True,
+                    )
+
+    def check_next_value_results(self, column, results):
```

Comment:
As with `check_previous_value_results`, feels like a lot of this can be refactored to use logic in `check_next_event_indexer`.